client/state/data-layer/wpcom/jetpack/connection/owner/index.js

```diff
@@ -0,0 +1,56 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { dispatchRequestEx } from 'state/data-layer/wpcom-http/utils';
+import { errorNotice, successNotice } from 'state/notices/actions';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { JETPACK_CONNECTION_CHANGE_OWNER } from 'state/action-types';
+import { requestJetpackUserConnectionData } from 'state/jetpack/connection/actions';
+
+const changeConnectionOwner = action =>
+	http(
+		{
+			apiVersion: '1.1',
+			method: 'POST',
+			path: '/jetpack-blogs/' + action.siteId + '/rest-api/',
+			body: {
+				path: '/jetpack/v4/connection/owner/',
+				body: JSON.stringify( { owner: action.newOwnerDotorgId } ),
```

Comment:
`newOwnerDotorgId` and `newOwnerWpcomDisplayName` feel a bit long - I'd probably get rid of the `newOwner` part because that part feels obvious. 

Also, my preference would be to use `wporgId` instead of `dotorgId` - we're already using `wpcom` and `wporg` here and there (when querying themes and plugins for example).

Reply:
> Also, my preference would be to use wporgId instead of dotorgId - we're already using wpcom and wporg

Yes, agree, I'll change that. I'd rather keep the `newOwner` part just to differentiate from current user ID.